### PR TITLE
Give unique names to action directionless parameters

### LINKED ISSUE #2710: Name collisions during inlining controls
Not sure if this has already been discussed, but there seems to be an issue with the `InlineDriver` pass.

Here is an example,

The control 'Wrapped' code,

```
control Wrapped(inout bit<8> valueSet) {
    @name("set") action set_0(bit<8> value) {
        valueSet = value;
    }
    @name("doSet") table doSet_0 {
        actions = {
            set_0();
        }
        default_action = set_0(8w0);
    }
    apply {
        doSet_0.apply();
    }
}
```
gets inlined into the control 'Wrapper'

```
control Wrapper(inout bit<8> value) { 
    @name("wrapped") Wrapped() wrapped_0; 
    apply { wrapped_0.apply(value); } 
}
```
as,

```
control Wrapper(inout bit<8> value) {
    @name("wrapped") Wrapped() wrapped_0;
    @name("wrapped.set") action wrapped_set(bit<8> value) {
        value = value;
    }
    @name("wrapped.doSet") table wrapped_doSet {
        actions = {
            wrapped_set();
        }
        default_action = wrapped_set(8w0);
    }
    apply {
        {
            wrapped_doSet.apply();
        }
    }
}
```
'value' from control param collides with local 'value' inside action param and creates an invalid
```
    @name("wrapped.set") action wrapped_set(bit<8> value) {
        value = value;
    }
```
This gets optimized out eventually which is incorrect.

Does the pass need a fix to create new variable / param names for such collisions?

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Fixes #2710
The code changes are small, but most reference outputs are changed.
I have also modified the test script to make sure that when new reference outputs are changed the p4info files do not change.

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
This was built on #2713

File: backends/p4test/run-p4-sample.py [hunk 1]
```diff
@@ -119,7 +119,8 @@ def target():
 timeout = 10 * 60
 
 def compare_files(options, produced, expected, ignore_case):
-    if options.replace:
+    # p4info files should not change
+    if options.replace and "p4info" not in produced:
```
Comment (fruffy) [inline backends/p4test/run-p4-sample.py:123]: Is the check also implemented for the other back end `run-p4-sample.py` files?

Comment (mbudiu-vmw) [reply backends/p4test/run-p4-sample.py:123]: I think that only this script generates and compares the p4info files.
This is the only script that generates and compares the reference output files.

Comment (fruffy) [reply backends/p4test/run-p4-sample.py:123]: Oh I see. So when you have files such as `xyz-bmv2.p4` tests are run twice. Once with p4test and the other time with the bmv2 script.

**Review by fruffy — APPROVED** [review]

**Pull request merged**